component/file_cache/file_cache.go

```diff
@@ -691,9 +692,47 @@ func (fc *FileCache) DeleteFile(options internal.DeleteFileOptions) error {
 	return nil
 }
 
-// OpenFile: Makes the file available in the local cache for further file operations.
-func (fc *FileCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Handle, error) {
-	log.Trace("FileCache::OpenFile : name=%s, flags=%d, mode=%s", options.Name, options.Flags, options.Mode)
+func (fc *FileCache) getHandleData(handle *handlemap.Handle) *handlemap.Handle {
+
+	log.Debug("FileCache::getHandleData : Need to download %s", handle.Path)
+
+	fc.mu.Lock()
+	//exctract the flags out of handle values
+	flags, _ := handle.GetValue("flag")
+	flagsStruct, ok := flags.(struct{ flags int })
+	if !ok {
+		log.Debug("FileCache::getHandleData : error Type assertion failed on getting flag for %s [%s]", handle.Path)
+		return nil
+	}
+
+	// Extract the integer as a separate variable
+	flag := flagsStruct.flags
+
+	// extract the filemode out of handle values
+	mode, _ := handle.GetValue("mode")
+	fileModeValue, ok := mode.(os.FileMode)
+	if !ok {
+		log.Debug("FileCache::getHandleData : error Type assertion failed on getting file mode for %s [%s]", handle.Path)
+		return nil
+	}
+
+	handle, err := fc.DownloadFile(internal.DownloadFileOptions{Name: handle.Path, Flags: flag, Mode: fileModeValue})
+
+	if err != nil {
+		log.Err("FileCache::getHandleData : error downloading data for file %s [%s]", handle.Path, err.Error())
+		return handle
+	}
+
+	handle.RemoveValue("flag")
+	handle.RemoveValue("mode")
+
+	fc.mu.Unlock()
+
+	return handle
+}
+
+func (fc *FileCache) DownloadFile(options internal.DownloadFileOptions) (*handlemap.Handle, error) {
```

Comment:
You could pass the whole handle into this function, and then you could modify its fields instead of replacing the whole struct.